fs/dafsserver/dafsserver.cpp

```diff
@@ -2655,6 +2656,255 @@ class CRemoteDiskWriteActivity : public CRemoteWriteBaseActivity
+
+        if (compression == "default")
+        {
+            flags |= HTREE_COMPRESSED_KEY;
+            compression = "";
+        }
+        else if (compression == "lzw")
+        {
+            flags |= HTREE_COMPRESSED_KEY;
+            compression = "";
+        }
+        else if (compression == "row")
+        {
+            compression = "";
+            flags |= HTREE_COMPRESSED_KEY | HTREE_QUICK_COMPRESSED_KEY;
+        }
+        else if (compression.substr(0,7) == "inplace")
+        {
+            flags |= HTREE_COMPRESSED_KEY;
+        }
+
+        bool isVariable = outMeta->isVariableSize();
+        if (isVariable)
+            flags |= HTREE_VARSIZE;
+
+        helper.setown(new CRemoteIndexWriteHelper(fileName.get(), compression.c_str(), inMeta, outMeta, flags));
+
+        unsigned nodeSize = NODESIZE;
+        if (config.hasProp("nodeSize"))
+        {
+            nodeSize = config.getPropInt("nodeSize");
+            helper->setIndexMeta("_nodeSize", std::to_string(nodeSize));
+        }
+
+        if (config.hasProp("noSeek"))
+        {
+            bool noSeek = config.getPropBool("noSeek");
+            helper->setIndexMeta("_noSeek", noSeek ? "true" : "false");
+            if (noSeek)
+                flags |= TRAILING_HEADER_ONLY;
+        }
+        
+        if (config.hasProp("useTrailingHeader"))
+        {
+            bool useTrailingHeader = config.getPropBool("useTrailingHeader");
+            helper->setIndexMeta("_useTrailingHeader", useTrailingHeader ? "true" : "false");
+            if (useTrailingHeader)
+                flags |= USE_TRAILING_HEADER;
+            else
+                flags &= ~USE_TRAILING_HEADER; 
+        }
+
+        size32_t fileposSize = hasTrailingFileposition(helper->queryDiskRecordSize()->queryTypeInfo()) ? sizeof(offset_t) : 0;
+        if (isVariable)
+        {
+            if (helper->getFlags() & TIWmaxlength)
+                maxDiskRecordSize = helper->getMaxKeySize();
+            else
+                maxDiskRecordSize = KEYBUILD_MAXLENGTH; // Current default behaviour, could be improved in the future
+        }
+        else
+            maxDiskRecordSize = helper->queryDiskRecordSize()->getFixedSize()-fileposSize;
+
+        if (maxDiskRecordSize > KEYBUILD_MAXLENGTH)
+            throw MakeStringException(99, "Index maximum record length (%d) exceeds 32k internal limit", maxDiskRecordSize);
+
+        rowBuffer.allocateN(maxDiskRecordSize, true);
+        prevRowBuffer.allocateN(maxDiskRecordSize, true);
+
+        openFileStream();
+        builder.setown(createKeyBuilder(iFileIOStream.get(), flags, maxDiskRecordSize, nodeSize, helper->getKeyedSize(), 0, helper.get(), compression.c_str(), true, false));
+    }
+
+    ~CRemoteIndexWriteActivity()
+    {
+        if (builder != nullptr && helper != nullptr)
+        {
+            Owned<IPropertyTree> metadata;
+            metadata.setown(createPTree("metadata", ipt_fast));
```

Comment:
trivial: ^ could be on 1 line : Owned<IPropertyTree> metadata = createPTree("metadata");

not worth diverging away from default to specify ipt_fast in this case, it's the default anyway.